crates/core_arch/src/x86/test.rs

```diff
@@ -139,3 +139,47 @@ pub unsafe fn assert_eq_m512d(a: __m512d, b: __m512d) {
         panic!("{:?} != {:?}", a, b);
     }
 }
+
+#[cfg(all(target_arch = "x86_64", windows))]
+#[target_feature(enable = "amx-tile")]
+pub(crate) fn _init_amx() {}
+
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+#[target_feature(enable = "amx-tile")]
+#[inline]
+pub(crate) unsafe fn _init_amx() {
+    let mut ret: usize;
+    let mut xfeatures: usize = 0;
+
+    core::arch::asm!(
+        "mov rax, 158",
+        "mov rdi, 0x1022",
+        "mov rsi, {p}",
+        "syscall",
+        out("rax") ret,
+        p = in(reg) &mut xfeatures,
+        options(nostack, preserves_flags),
+    );
+    if ret != 0 {
+        panic!("Failed to get XFEATURES");
+    } else {
+        match 0b11 & (xfeatures >> 17) {
+            0 => panic!("AMX is not available"),
+            1 => {
+                core::arch::asm!(
+                    "mov rax, 158",
+                    "mov rdi, 0x1023",
+                    "mov rsi, 18",
+                    "syscall",
+                    out("rax") ret,
+                    options(nostack, preserves_flags),
+                );
+                if ret != 0 {
+                    panic!("Failed to enable AMX");
+                }
+            }
+            3 => {}
+            _ => unreachable!(),
+        }
+    }
+}
```

Comment:
See [Intel's sample code](https://github.com/intel/AMX-TMUL-Code-Samples/blob/main/src%2Ftest-amxtile.c#L85-L99). They use only one syscall to directly set xtiledata without checking for xtilecfg (as xtiledata requires xtilecfg, that call would simply fail if xtilecfg is not enabled). Also, as you are only using it in one file (`x86_64/amx.rs`), it makes sense to move these functions to the `tests` module of amx.rs

Reply:
The first syscall is arch_prctl(ARCH_GET_XCOM_PERM) which is a read-only operation and has 2~3x less overhead compared with the latter syscall arch_prctl(ARCH_REQ_XCOMP_PERM), which might execute a write operation. The `xtiledata` flag only needs to be set once for a process during its lifetime. So we can apply the first "read" syscall to check if there is a need to follow up another "request" syscall in case the xtiledata has already been turned on.